src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -409,7 +409,8 @@ ostree_repo_static_delta_execute_offline_with_signature (OstreeRepo *self, GFile
         return glnx_throw_errno_prefix (error, "openat(O_DIRECTORY)");
       else
         {
-          g_autofree char *dir = dirname (g_strdup (dir_or_file_path));
+          g_autofree char *dir_or_file_path_copy = g_strdup (dir_or_file_path);
+          g_autofree char *dir = dirname (dir_or_file_path_copy);
```

Comment:
[dirname](https://man7.org/linux/man-pages/man3/dirname.3.html) doesn't make a copy of the passed in string, so this is almost certainly going to be a double free with the second `g_autofree`. Since the string is already being copied, it would probably me more straightforward to use [g_path_get_dirname](https://docs.gtk.org/glib/func.path_get_dirname.html) like the `g_path_get_basename` right below this. That API does what you'd expect by not modifying the string in place and coverity should be able to understand it.